python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -183,6 +183,82 @@ def __init__(self, experiment_checkpoint_path, trials=None):
         super(ExperimentAnalysis, self).__init__(
             os.path.dirname(experiment_checkpoint_path))
 
+    def get_best_trial(self, metric, mode="max", scope="all"):
+        """Retrieve the best trial object. Compare all trials' scores on
+            `metric`, based on `mode`. Only factor in `scope` steps.
+
+        Args:
+            metric (str): Key for trial info to order on.
+            mode (str): One of [min, max].
+            scope (str): One of [all, last].
+        """
+        if mode not in ["max", "min"]:
+            logger.warning(
+                "ExperimentAnalysis: attempting to get best trial for "
+                "metric {} for mode {} not in [\"max\", \"min\"]".format(
+                    metric, mode))
+            return None
+        if scope not in ["all", "last"]:
+            logger.warning(
+                "ExperimentAnalysis: attempting to get best trial for "
+                "metric {} for scope {} not in [\"all\", \"last\"]".format(
+                    metric, scope))
+            return None
+        best_trial = None
+        best_metric_score = None
+        for trial in self.trials:
+            if metric not in trial.metric_analysis.keys():
+                continue
+
+            if mode == "max":
+                metric_score = trial.metric_analysis[metric]["max"]
+            elif mode == "min":
+                metric_score = trial.metric_analysis[metric]["min"]
+            else:
```

Comment:
mode is always in `[max, min]` right?

think this can just be:
`metric_score = trial.metric_analysis[metric][mode]`